packages/core/src/render3/jit/directive.ts

```diff
@@ -140,7 +140,7 @@ export function compileComponent(type: Type<any>, metadata: Component): void {
             // Patch the component definition for standalone components with `directiveDefs` and
             // `pipeDefs` functions which lazily compute the directives/pipes available in the
             // standalone component. Also set `dependencies` to the lazily resolved list of imports.
-            const imports: Type<any>[] = flatten(metadata.imports || EMPTY_ARRAY);
+            const imports: Type<any>[] = flatten([...(metadata.imports ?? EMPTY_ARRAY)]);
```

Comment:
What is going on here is that the `flatten` function _must_ work with mutable arrays (it is used in queries where it is required for perf / correctness). 

We might want to introduce another, lighter version of flatten for JiT?

Reply:
nit: looking at the `flatten` implementation and given the fact that we end up creating a new array here anyways, I think we can just pass an empty array as the second argument, WDYT?

```suggestion
            const imports: Type<any>[] = flatten(metadata.imports ?? EMPTY_ARRAY, []);
```